### PR TITLE
Training update

### PR DESCRIPTION
**Pull request by SergioGasquez**
Sorry for the long PR, here is a summary:
- Pin crates version to latest released version
  - Updated examples accordingly
  - Remove unnecessary dependencies
  - Updated the book to match the updated examples
- Update nightly version  
- LanguageTool suggestions
- Delete BSC
  - Created 2 crates: One for wifi and another for ws2812
- Updated container
- Removed `icm42670p` crate, we now use https://crates.io/crates/icm42670 instead

### CODE REVIEW

File: intro/http-client/examples/https_client.rs [hunk 1]
```diff
@@ -21,63 +21,65 @@ pub struct Config {
...
+    let connection = EspHttpConnection::new(&Configuration {
         use_global_ca_store: true,
         crt_bundle_attach: Some(esp_idf_sys::esp_crt_bundle_attach),
-
         ..Default::default()
     })?;
+    let mut client = Client::wrap(connection);
 
-    // 2. open a GET request to `url`
+    // // 2. Open a GET request to `url`
```
Comment (JurajSadel) [inline intro/http-client/examples/https_client.rs:56]: Probably copy-paste typo, can we remove one of `//`?

File: intro/http-client/examples/https_client.rs [hunk 2]
```diff
@@ -21,63 +21,65 @@ pub struct Config {
...
-    // (since we don't send anything - but have to do the writer step anyway)
-    //
-    // https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/esp_http_client.html
-    // if this were a POST request, you'd set a write length > 0 and then writer.do_write(&some_buf);
-    let writer = request.into_writer(0)?;
-    // 4. submit our write request and check the status code of the response.
-    // Successful http status codes are in the 200..=299 range.
-    let response = writer.into_response()?;
+    // // 3. Submit write request and check the status code of the response.
+    // // Successful http status codes are in the 200..=299 range.
```
Comment (JurajSadel) [inline intro/http-client/examples/https_client.rs:60]: Same as above

**Review by JurajSadel — APPROVED** [review]
LGTM! I left a few not-that-important comments.

File: book/src/01_intro.md [hunk 3]
```diff
@@ -12,18 +12,18 @@ You can join the [esp-rs community](https://matrix.to/#/#esp-rs:matrix.org) on M
 
 ## The board
 
-An [Espressif Rust Board](https://github.com/esp-rs/esp-rust-board) is mandatory[^note] for working with this book - emulators like QEMU are not supported.
+A [Rust ESP Board](https://github.com/esp-rs/esp-rust-board) is mandatory[^note] for working with this book - emulators like QEMU are not supported.
```
[the remark this answers is not in the document]
Comment (SergioGasquez) [reply book/src/01_intro.md:15]: Thanks for the review! At the moment the Wokwi examples are pretty outdate and not working. Just created https://github.com/esp-rs/espressif-trainings/issues/154 to update them and support Wokwi VsCode extension, as part of that issue we can update it.

**Review by MabezDev — APPROVED** [review]
A lot to digest here, but everything looks okay from a couple of passes of the changes. Thanks for taking care of this!

**Pull request merged**